avocado/plugins/resolvers.py

```diff
@@ -19,15 +19,26 @@
 import os
 import re
 
+from avocado.core import data_dir
 from avocado.core.nrunner import Runnable
-from avocado.core.plugin_interfaces import Resolver
+from avocado.core.plugin_interfaces import Discoverer, Resolver
 from avocado.core.references import reference_split
 from avocado.core.resolver import (ReferenceResolution,
-                                   ReferenceResolutionResult, check_file)
+                                   ReferenceResolutionResult, check_file,
+                                   extend_directory)
 from avocado.core.safeloader import find_avocado_tests, find_python_unittests
 from avocado.core.settings import settings
 
 
+def discover(resolver):
+    resolutions = []
+    config = settings.as_dict()
+    if config.get('list.resolver'):
+        for reference in extend_directory(data_dir.get_test_dir()):
```

Comment:
I understand that today, we have a little bit of inconsistency with regards to those dirs, but I'm trying to remove them (vide #4443).

What happen if you use  `config.get("datadir.paths.test_dir")` here?

Reply:
I reused the solution from legacy runner. But yes I can change that to `config.get("datadir.paths.test_dir")`